### PR TITLE
S2198: Add C#

### PR DESCRIPTION
**Pull request by gregory-paidis-sonarsource**
Rule implementation: https://github.com/SonarSource/sonar-dotnet/issues/6658

### CODE REVIEW

File: rules/S2198/csharp/rule.adoc [hunk 1]
```diff
@@ -1,4 +1,21 @@
-include::../rule.adoc[]
+Certain mathematical comparisons will always return the same value, and should simply not be made. 
+
+These comparisons will always return ``++false++``:
+
+* comparing a ``++char++`` with a numeric constant that's outside the ``++char++`` range
+* comparing a ``++float++`` with a numeric constant that's outside the ``++float++`` range
+* comparing a ``++long++`` with a numeric constant that's outside the ``++long++`` range
+* comparing a ``++ulong++`` with a numeric constant that's outside the ``++ulong++`` range
```
Comment (martin-strecker-sonarsource) [inline rules/S2198/csharp/rule.adoc:8]:
```suggestion
These comparisons will always return ``++false++``:

* comparing a `char` with a numeric constant that's outside the `char` range
* comparing a `float` with a numeric constant that's outside the `float` range
* comparing a `long` with a numeric constant that's outside the `long` range
* comparing a `ulong` with a numeric constant that's outside the `ulong` range
```

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]

File: rules/S2198/csharp/rule.adoc [hunk 2]
```diff
@@ -1,4 +1,21 @@
-include::../rule.adoc[]
+Certain mathematical comparisons will always return the same value, and should simply not be made. 
+
+These comparisons will always return ``++false++``:
```
Comment (martin-strecker-sonarsource) [inline rules/S2198/csharp/rule.adoc:3]:
```suggestion
These comparisons will always return `false`:
```

File: rules/S2198/csharp/rule.adoc [hunk 3]
```diff
@@ -1,4 +1,21 @@
...
+
+* comparing a `char` with a numeric constant that's outside the `char` range
```
Comment (martin-strecker-sonarsource) [inline rules/S2198/csharp/rule.adoc:5]: The convention is to start with an upper letter and to end with a dot:
```suggestion
* Comparing a `char` with a numeric constant that's outside the `char` range.
```

File: rules/S2198/csharp/rule.adoc [hunk 4]
```diff
@@ -1,4 +1,21 @@
...
+* comparing a `long` with a numeric constant that's outside the `long` range
+* comparing a `ulong` with a numeric constant that's outside the `ulong` range
+
+== Noncompliant Code Example
+
+[source,text]
+----
+float f = 42.0f;
+const double d = double.MaxValue;
+if (f <= d) { } // Noncompliant
```
Comment (martin-strecker-sonarsource) [inline rules/S2198/csharp/rule.adoc:16]: Simpler:
```suggestion
float f = 42.0f;
if (f <= double.MaxValue) { } // Noncompliant
```

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]

File: rules/S2198/csharp/rule.adoc (hunk 3 above)
Comment (martin-strecker-sonarsource) [inline rules/S2198/csharp/rule.adoc:5]: Maybe better like so?
```suggestion
* Comparing a `char` with a numeric constant that is outside of the range of `char`.
```

File: rules/S2198/csharp/rule.adoc [hunk 5]
```diff
@@ -1,4 +1,20 @@
-include::../rule.adoc[]
+Certain mathematical comparisons will always return the same value, and should simply not be made. 
+
+These comparisons will always return `false`:
```
Comment (martin-strecker-sonarsource) [inline rules/S2198/csharp/rule.adoc:3]:
```suggestion
These comparisons will always return the same value (either always `true` or always `false` depending on the kind of comparison):
```

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]
"always false" is not correct.

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM. Please merge once 🟢 .

**Comment by arseniy-sonar** [comment]
Hi @gregory-paidis-sonarsource 
Please, follow [the guideline](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule) for the commit and the PR summary next time.

**Pull request merged**